middle_end/flambda2/to_cmm/to_cmm_expr.ml

```diff
@@ -265,15 +267,39 @@ let translate_apply0 ~dbg_with_inlined:dbg env res apply =
     in
     let effects = To_cmm_effects.transl_c_call_effects effects in
     let coeffects = To_cmm_effects.transl_c_call_coeffects coeffects in
-    ( wrap dbg
-        (C.extcall ~dbg ~alloc:needs_caml_c_call ~is_c_builtin ~effects
-           ~coeffects ~returns ~ty_args callee
-           (C.Extended_machtype.to_machtype return_ty)
-           args),
-      free_vars,
-      env,
-      res,
-      Ece.all )
+    let extcall =
+      C.extcall ~dbg ~alloc:needs_caml_c_call ~is_c_builtin ~effects ~coeffects
+        ~returns ~ty_args callee return_ty args
+    in
+    let wrap return_values =
+      let kinds = Flambda_arity.unarized_components return_arity in
+      assert (List.compare_length_with kinds (Array.length component_tys) = 0);
```

Comment:
Does that assertion match the comment above about `void` ? I.e. when exactly does `void` disappear from return arity & others unboxed tuples ?

Reply:
I think it should do.  Have added a comment.  We'll get tests for void in due course (unsure exactly when).